src/Config/ObjectModel/HostOptions.cs

```diff
@@ -1,6 +1,78 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+using System.Diagnostics.CodeAnalysis;
+using System.Net;
+using System.Text.Json.Serialization;
+using Azure.DataApiBuilder.Service.Exceptions;
+
 namespace Azure.DataApiBuilder.Config.ObjectModel;
 
-public record HostOptions(CorsOptions? Cors, AuthenticationOptions? Authentication, HostMode Mode = HostMode.Production);
+public record HostOptions
+{
+    /// <summary>
+    /// Dab engine can at maximum handle 158 MB of data in a single response from a source.
+    /// Json deserialization of a response into a string has a limit of 166,666,666 bytes which when converted to MB is 158 MB.
+    /// ref: enforcing code: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.cs#L103
+    /// ref: Json constant: https://github.com/stephentoub/runtime/blob/main/src/libraries/System.Text.Json/src/System/Text/Json/JsonConstants.cs
```

Comment:
nit: https://github.com/dotnet/runtime link is for latest runtime code, you'd want to ensure you select tag for .net8 and/or .net6 to demonstrate the code responsible for behavior we see today.

nit: https://github.com/stephentoub is .net engineer's personal repo `System.Text.Json/src/System/Text/Json/JsonConstants.cs`.. any chance you can make this direct link to dotnet/runtime repo?

Reply:
Yes let me fix this

Reply:
fixed.